tests/components/create_post/create_post.test.jsx

```diff
@@ -0,0 +1,642 @@
+        instance.handleKeyDown({keyCode: Constants.KeyCodes.UP, preventDefault: jest.fn()});
+        expect(setEditingPost).toHaveBeenCalledWith(currentUsersLatestPostProp.id, commentCountForPostProp, '#post_textbox', type);
+    });
+
+    it('Should call edit action as post for arrow up', () => {
+        const setEditingPost = jest.fn();
+        const wrapper = shallow(createPost({
+            actions: {
+                ...actionsProp,
+                setEditingPost
+            }
+        }));
+        const instance = wrapper.instance();
+
+        wrapper.setProps({
+            currentUsersLatestPost: {id: 'b', channel_id: currentChannelProp.id}
+        });
+
+        const type = Utils.localizeMessage('create_post.post', Posts.MESSAGE_TYPES.POST);
+        instance.handleKeyDown({keyCode: Constants.KeyCodes.UP, preventDefault: jest.fn()});
+        expect(setEditingPost).toHaveBeenCalledWith(currentUsersLatestPostProp.id, commentCountForPostProp, '#post_textbox', type);
+    });
+
+    it('Should call moveHistoryIndexForward as ctrlKey and down arrow', () => {
+        const moveHistoryIndexForward = jest.fn(
+            () => {
+                return new Promise((resolve) => {
+                    process.nextTick(() => resolve());
+                });
+            }
+        );
+        const wrapper = shallow(createPost({
+            actions: {
+                ...actionsProp,
+                moveHistoryIndexForward
+            }
+        }));
+        const instance = wrapper.instance();
+
+        instance.handleKeyDown({keyCode: Constants.KeyCodes.DOWN, ctrlKey: true, preventDefault: jest.fn()});
+        expect(moveHistoryIndexForward).toHaveBeenCalled();
+    });
+
+    it('Should call moveHistoryIndexBack as ctrlKey and up arrow', () => {
+        const moveHistoryIndexBack = jest.fn(
+            () => {
+                return new Promise((resolve) => {
+                    process.nextTick(() => resolve());
+                });
+            }
+        );
+        const wrapper = shallow(createPost({
+            actions: {
+                ...actionsProp,
+                moveHistoryIndexBack
+            }
+        }));
+        const instance = wrapper.instance();
+
+        instance.handleKeyDown({keyCode: Constants.KeyCodes.UP, ctrlKey: true, preventDefault: jest.fn()});
+        expect(moveHistoryIndexBack).toHaveBeenCalled();
+    });
+
+    it('Show tutorial', () => {
+        const wrapper = shallow(createPost({
+            showTutorialTip: 1
+        }));
+        expect(wrapper.find('TutorialTip').length).toBe(1);
+    });
+
+    it('Toggle showPostDeletedModal state', () => {
+        const wrapper = shallow(createPost());
+        const instance = wrapper.instance();
+        instance.showPostDeletedModal();
+        expect(wrapper.state('showPostDeletedModal')).toBe(true);
+
+        instance.hidePostDeletedModal();
+        expect(wrapper.state('showPostDeletedModal')).toBe(false);
+    });
+});
```

Comment:
Kudos to tests! There are some cases to add but it's good to have these tests for now.